ntbs-service/Services/NotificationSearchBuilder.cs

```diff
@@ -33,7 +34,7 @@ public INotificationSearchBuilder FilterById(string id)
             {
                 int.TryParse(id, out int parsedId);
                 notificationIQ = notificationIQ.Where(s => s.NotificationId.Equals(parsedId) 
-                    || s.ETSID.Equals(id) || s.LTBRID.Equals(id) || s.PatientDetails.NhsNumber.Equals(id));
+                    || (s.ETSID != null && s.ETSID.Equals(id)) || (s.LTBRID != null && s.LTBRID.Equals(id)) || s.PatientDetails.NhsNumber.Equals(id));

```

Comment:
This is needed for tests to pass (a nullPointerException is thrown otherwise) however is not necessary outside of testing - I will create a tech debt ticket to look into this further as I haven't figured out exactly why this is yet.